src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs

```diff
@@ -216,24 +215,23 @@ public static IServiceCollection ConfigureExtensions(this IServiceCollection ser
         {
             // Add the services to discover extensions
             services.AddSingleton<ExtensionDiscoverer>();
+            services.TryAddSingleton<IDotnetToolsFileSystem, DefaultDotnetToolsFileSystem>();
```

Comment:
I think all of the changes in this file should be undone _except_ for this line. The folder location that is next to the assembly shouldn't be configurable by test IMO. Like I mentioned before, testing the user and shared locations shares an equivalency with the "next to assembly" location, so it doesn't need to be explicitly tested, which simplifies the override knobs.